test/integration/serviceaccount/service_account_test.go

```diff
@@ -371,18 +371,26 @@ func TestLegacyServiceAccountTokenTracking(t *testing.T) {
 			myConfig.BearerToken = string(test.secretTokenData)
 			roClient := clientset.NewForConfigOrDie(&myConfig)
 			dateBefore := time.Now().UTC().Format(dateFormat)
-			go func() {
-				doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
-			}()
-			doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
+
+			var wg sync.WaitGroup
+			concurrency := 5
+			for i := 0; i < concurrency; i++ {
+				wg.Add(1)
+				go func() {
+					doServiceAccountAPIRequests(t, roClient, myns, true, true, false)
+					wg.Done()
+				}()
+			}
+			wg.Wait()
```

Comment:
this was the flaky bit? we weren't waiting for the async doServiceAccountAPIRequests invocation to finish before?

Reply:
yea, forgot it :(